src/megameklab/com/printing/RecordSheetTask.java

```diff
@@ -105,6 +105,8 @@ protected void done() {
         try {
             get();
         } catch (ExecutionException e) {
+            JOptionPane.showMessageDialog(null, e.getMessage(),
+                    "A problem has occurred", JOptionPane.ERROR_MESSAGE);
```

Comment:
This doesn't fix issue #1002.
The change adds a dialog for a captured hidden exception, which we don't need as it's handled and logged already.
However, #1002 is for a fully uncaught exception, and thus is caused by either missing a catch or by being outside of try catch logic altogether.

Reply:
Im not sure what you mean. Did you test it? Did the notice window not appear when you wrote to a locked PDF? Are you thinking of another fix (I can't think of one, when the file is locked it's locked).
Whatever my issue 1002 that I wrote up in 28s just as a reminder stated, when writing to a locked PDF the user just sees it doing nothing. The general catcher that you put in two weeks ago doesnt seem to catch it. The log shows an exception which (thanks to your log4j xml) went to my console. I wrote uncaught but thats nonsense, it is caught but just logged. What I did is add the notice dialog which appears as it's supposed to do giving a hint to the user.

Reply:
Ah, it's been a miscommunication from the start.

One of my recent logging changes adds uncaught exception handling, which write to the log with the following popup message.
![unknown](https://user-images.githubusercontent.com/39067288/151702423-d4996dbd-076c-463e-b20c-88b16fb59301.png)

I thought you were referring to this setup, in which case my comments would be correct.

However, given this is not because of that setup my previous comment is completely incorrect, and this looks fine.